### ISSUE TITLE
Misleading method name with "magic" fallback

### ISSUE THREAD

**Issue by polamayster** [oauthlib/oauthlib#718]
https://github.com/oauthlib/oauthlib/blob/b71636e85f845b79b5a56de6480fcba9d1415720/oauthlib/oauth2/rfc6749/tokens.py#L240-L257
It would be much more predictable/cleaner if method either did not fallback to body payload/uri_query on line 255:
https://github.com/oauthlib/oauthlib/blob/b71636e85f845b79b5a56de6480fcba9d1415720/oauthlib/oauth2/rfc6749/tokens.py#L254-L255
or method was renamed to reflect what it actually does.

**Comment by JonathanHuot** [comment]
Yes you're right, I think this part of the code deserve a refactoring to allow https://github.com/oauthlib/oauthlib/issues/609 , which is essential .
